### Matthew Riddell 
### D00245674
### MSc in Data Analytics
### Dissertation
### Initial Data Exploration Jupyter Notebook

In this part, I will conduct Initial EDA of the scraped ECI petition datasets and the imported Worldstat dataset

For this part I will investigate the data itself i.e. dataset size and shape, the variable types, missing values, duplicates etc so that I can plan on how to properly merge the datasets and clean them for proper Data Analysis 

Data Exploration conducted before Data Preparation and Cleaning

### Data Description

The extracted variables from each The Petition Datasets:

- Country Name
- Number of Signatures
- Country Minimum Threshold
- Percentage of Threshold

These variables from the Worldstat Dataset will be selected:

- Country Name
- Population
- GDP
- Unemployment Rate
- Life Expectancy
- Population Density
- Gross Tertiary Education Enrollment
- Urban Population
- Labor Participation

### Link to each Dataset:

### ECI Petitions:
#### 2024

- ECI 01: https://citizens-initiative.europa.eu/initiatives/details/2024/000001_en Ban on conversion practices in the European Union
- ECI 02: https://citizens-initiative.europa.eu/initiatives/details/2024/000002_en European Citizens' Initiative in Defence of Agriculture and Rural Economy in Europe
- ECI 03: https://citizens-initiative.europa.eu/initiatives/details/2024/000004_en My Voice, My Choice: For Safe And Accessible Abortion
- ECI 04: https://citizens-initiative.europa.eu/initiatives/details/2024/000006_en Air-Quotas
- ECI 05: https://citizens-initiative.europa.eu/initiatives/details/2024/000007_en Stop Destroying Videogames
- ECI 06: https://citizens-initiative.europa.eu/initiatives/details/2024/000008_en Stop Cruelty Stop Slaughter
- ECI 07: https://citizens-initiative.europa.eu/initiatives/details/2024/000009_en STOP FAKE FOOD: ORIGIN ON LABEL
- ECI 08: https://citizens-initiative.europa.eu/initiatives/details/2024/000010_en ECI for a Water-Smart and Resilient Europe
- ECI 09: https://citizens-initiative.europa.eu/initiatives/details/2024/000011_en PsychedeliCare

#### 2023

- ECI 10: https://citizens-initiative.europa.eu/initiatives/details/2023/000001_en Stop torture and inhuman treatment at Europe’s borders
- ECI 11: https://citizens-initiative.europa.eu/initiatives/details/2023/000002_en Ensuring a dignified reception of migrants in Europe
- ECI 12: https://citizens-initiative.europa.eu/initiatives/details/2023/000003_en End The Horse Slaughter Age
- ECI 13: https://citizens-initiative.europa.eu/initiatives/details/2023/000004_en Connecting all European capitals and people through a high-speed train network
- ECI 14: https://citizens-initiative.europa.eu/initiatives/details/2023/000005_en Effective implementation of the concept of judicial precedent in EU countries
- ECI 15: https://citizens-initiative.europa.eu/initiatives/details/2023/000006_en Taxing great wealth to finance the ecological and social transition
- ECI 16: https://citizens-initiative.europa.eu/initiatives/details/2023/000007_en Preservation and development of Ukrainian culture, education, language, and traditions in EU states
- ECI 17: https://citizens-initiative.europa.eu/initiatives/details/2023/000008_en EU Live Bus Stop Info
- ECI 18: https://citizens-initiative.europa.eu/initiatives/details/2023/000009_en Trust and Freedom
- ECI 19: https://citizens-initiative.europa.eu/initiatives/details/2023/000011_en Creation of a European Environment Authority

#### 2022

- ECI 20: https://citizens-initiative.europa.eu/initiatives/details/2022/000001_en Win It On The Pitch 
- ECI 21: https://citizens-initiative.europa.eu/initiatives/details/2022/000002_en Fur Free Europe
- ECI 22: https://citizens-initiative.europa.eu/initiatives/details/2022/000003_en End The Slaughter Age
- ECI 23: https://citizens-initiative.europa.eu/initiatives/details/2022/000004_en Good Clothes, Fair Pay
- ECI 24: https://citizens-initiative.europa.eu/initiatives/details/2022/000007_en Protect the EU’s Rural Heritage, Food Security and Supply
- ECI 25: https://citizens-initiative.europa.eu/initiatives/details/2022/000008_en Focus on Specific Learning Disabilities on EU Level
- ECI 26: https://citizens-initiative.europa.eu/initiatives/details/2022/000009_en European Citizens’ Initiative for VEGAN MEAL
- ECI 27: https://citizens-initiative.europa.eu/initiatives/details/2022/000010_en European Day of "Whatever it Takes" 

### Worldstat Kaggle Dataset
https://www.kaggle.com/datasets/nelgiriyewithana/countries-of-the-world-2023

### Mahara Portfolio:
https://mahara.dkit.ie/view/view.php?t=f816cb6da6dd833aaa2f

### Github Repository:
https://github.com/Matthew-Riddell/MSc-Dissertation

In [227]:
import pandas as pd 
import numpy as np

#### File path and file loading

In [228]:
file_path = "C:/Users/Matty/Documents/College Notes & Assignments/Year 5/Dissertation/MSc-Dissertation/data/Unprepared Datasets/"

eci_01 = pd.read_csv(file_path + "eci_table_01.csv")
eci_02 = pd.read_csv(file_path + "eci_table_02.csv")
eci_03 = pd.read_csv(file_path + "eci_table_03.csv")
eci_04 = pd.read_csv(file_path + "eci_table_04.csv")
eci_05 = pd.read_csv(file_path + "eci_table_05.csv")
eci_06 = pd.read_csv(file_path + "eci_table_06.csv")
eci_07 = pd.read_csv(file_path + "eci_table_07.csv")
eci_08 = pd.read_csv(file_path + "eci_table_08.csv")
eci_09 = pd.read_csv(file_path + "eci_table_09.csv")
eci_10 = pd.read_csv(file_path + "eci_table_10.csv")
eci_11 = pd.read_csv(file_path + "eci_table_11.csv")
eci_12 = pd.read_csv(file_path + "eci_table_12.csv")
eci_13 = pd.read_csv(file_path + "eci_table_13.csv")
eci_14 = pd.read_csv(file_path + "eci_table_14.csv")
eci_15 = pd.read_csv(file_path + "eci_table_15.csv")
eci_16 = pd.read_csv(file_path + "eci_table_16.csv")
eci_17 = pd.read_csv(file_path + "eci_table_17.csv")
eci_18 = pd.read_csv(file_path + "eci_table_18.csv")
eci_19 = pd.read_csv(file_path + "eci_table_19.csv")
eci_20 = pd.read_csv(file_path + "eci_table_20.csv")
eci_21 = pd.read_csv(file_path + "eci_table_21.csv")
eci_22 = pd.read_csv(file_path + "eci_table_22.csv")
eci_23 = pd.read_csv(file_path + "eci_table_23.csv")
eci_24 = pd.read_csv(file_path + "eci_table_24.csv")
eci_25 = pd.read_csv(file_path + "eci_table_25.csv")
eci_26 = pd.read_csv(file_path + "eci_table_26.csv")
eci_27 = pd.read_csv(file_path + "eci_table_27.csv")

worldstat = pd.read_csv(file_path + "world-data-2023.csv")

### Dataset Shape

In [229]:
print("ECI 01:", eci_01.shape)
print("ECI 02:", eci_02.shape)
print("ECI 03:", eci_03.shape)
print("ECI 04:", eci_04.shape)
print("ECI 05:", eci_05.shape)
print("ECI 06:", eci_06.shape)
print("ECI 07:", eci_07.shape)
print("ECI 08:", eci_08.shape)
print("ECI 09:", eci_09.shape)
print("ECI 10:", eci_10.shape)
print("ECI 11:", eci_11.shape)
print("ECI 12:", eci_12.shape)
print("ECI 13:", eci_13.shape)
print("ECI 14:", eci_14.shape)
print("ECI 15:", eci_15.shape)
print("ECI 16:", eci_16.shape)
print("ECI 17:", eci_17.shape)
print("ECI 18:", eci_18.shape)
print("ECI 19:", eci_19.shape)
print("ECI 20:", eci_20.shape)
print("ECI 21:", eci_21.shape)
print("ECI 22:", eci_22.shape)
print("ECI 23:", eci_23.shape)
print("ECI 24:", eci_24.shape)
print("ECI 25:", eci_25.shape)
print("ECI 26:", eci_26.shape)
print("ECI 27:", eci_27.shape)
print("Worldstat:", worldstat.shape)

ECI 01: (28, 4)
ECI 02: (28, 4)
ECI 03: (28, 4)
ECI 04: (28, 4)
ECI 05: (28, 4)
ECI 06: (28, 4)
ECI 07: (28, 4)
ECI 08: (28, 4)
ECI 09: (28, 4)
ECI 10: (28, 4)
ECI 11: (28, 4)
ECI 12: (28, 4)
ECI 13: (28, 4)
ECI 14: (28, 4)
ECI 15: (28, 4)
ECI 16: (28, 4)
ECI 17: (28, 4)
ECI 18: (28, 4)
ECI 19: (28, 4)
ECI 20: (28, 4)
ECI 21: (28, 4)
ECI 22: (28, 4)
ECI 23: (28, 4)
ECI 24: (28, 4)
ECI 25: (28, 4)
ECI 26: (28, 4)
ECI 27: (28, 4)
Worldstat: (195, 35)


All 7 petitions contain the same overall structure with 28 rows each and 4 columns 

The worldstat dataset contains a much more complex structure with 195 rows and 35 columns

#### Display first 5 rows of each dataset

In [230]:
eci_01.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"10,724","13,395",80.06%
1,Belgium,"47,459","14,805",320.56%
2,Bulgaria,"2,386","11,985",19.91%
3,Croatia,"8,574","8,460",101.35%
4,Cyprus,696,"4,230",16.45%


In [231]:
eci_02.head()

,Country,Signatures,Threshold,Percentage
0,Austria,109,"13,395",0.81%
1,Belgium,160,"14,805",1.08%
2,Bulgaria,47,"11,985",0.39%
3,Croatia,76,"8,460",0.90%
4,Cyprus,27,"4,230",0.64%


In [232]:
eci_03.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"25,373","13,395",189.42%
1,Belgium,"18,313","14,805",123.69%
2,Bulgaria,"6,552","11,985",54.67%
3,Croatia,"65,963","8,460",779.70%
4,Cyprus,"3,228","4,230",76.31%


In [233]:
eci_04.head()

,Country,Signatures,Threshold,Percentage
0,Austria,156,"13,395",1.16%
1,Belgium,257,"14,805",1.74%
2,Bulgaria,17,"11,985",0.14%
3,Croatia,24,"8,460",0.28%
4,Cyprus,8,"4,230",0.19%


In [234]:
eci_05.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"20,714","13,395",154.64%
1,Belgium,"31,846","14,805",215.10%
2,Bulgaria,"14,238","11,985",118.80%
3,Croatia,"14,403","8,460",170.25%
4,Cyprus,"1,997","4,230",47.21%


In [235]:
eci_06.head()

,Country,Signatures,Threshold,Percentage
0,Austria,714,"13,395",5.33%
1,Belgium,564,"14,805",3.81%
2,Bulgaria,122,"11,985",1.02%
3,Croatia,472,"8,460",5.58%
4,Cyprus,29,"4,230",0.69%


In [236]:
eci_07.head()

,Country,Signatures,Threshold,Percentage
0,Austria,352,"13,395",2.63%
1,Belgium,426,"14,805",2.88%
2,Bulgaria,185,"11,985",1.54%
3,Croatia,217,"8,460",2.57%
4,Cyprus,63,"4,230",1.49%


In [237]:
eci_08.head()

,Country,Signatures,Threshold,Percentage
0,Austria,177,"13,395",1.32%
1,Belgium,217,"14,805",1.47%
2,Bulgaria,50,"11,985",0.42%
3,Croatia,64,"8,460",0.76%
4,Cyprus,20,"4,230",0.47%


In [238]:
eci_09.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"1,139","13,395",8.50%
1,Belgium,"2,993","14,805",20.22%
2,Bulgaria,197,"11,985",1.64%
3,Croatia,"1,115","8,460",13.18%
4,Cyprus,53,"4,230",1.25%


In [239]:
eci_10.head()

,Country,Signatures,Threshold,Percentage
0,Austria,200,"13,395",1.49%
1,Belgium,253,"14,805",1.71%
2,Bulgaria,36,"11,985",0.30%
3,Croatia,44,"8,460",0.52%
4,Cyprus,1,"4,230",0.02%


In [240]:
eci_11.head()

,Country,Signatures,Threshold,Percentage
0,Austria,100,"13,395",0.75%
1,Belgium,280,"14,805",1.89%
2,Bulgaria,21,"11,985",0.18%
3,Croatia,16,"8,460",0.19%
4,Cyprus,4,"4,230",0.09%


In [241]:
eci_12.head()

,Country,Signatures,Threshold,Percentage
0,Austria,194,"13,395",1.45%
1,Belgium,182,"14,805",1.23%
2,Bulgaria,84,"11,985",0.70%
3,Croatia,57,"8,460",0.67%
4,Cyprus,6,"4,230",0.14%


In [242]:
eci_13.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"1,133","13,395",8.46%
1,Belgium,"1,336","14,805",9.02%
2,Bulgaria,492,"11,985",4.11%
3,Croatia,798,"8,460",9.43%
4,Cyprus,33,"4,230",0.78%


In [243]:
eci_14.head()

,Country,Signatures,Threshold,Percentage
0,Austria,9,"13,395",0.07%
1,Belgium,15,"14,805",0.10%
2,Bulgaria,5,"11,985",0.04%
3,Croatia,2,"8,460",0.02%
4,Cyprus,2,"4,230",0.05%


In [244]:
eci_15.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"4,286","13,395",32.00%
1,Belgium,"13,942","14,805",94.17%
2,Bulgaria,227,"11,985",1.89%
3,Croatia,424,"8,460",5.01%
4,Cyprus,113,"4,230",2.67%


In [245]:
eci_16.head()

,Country,Signatures,Threshold,Percentage
0,Austria,54,"13,395",0.40%
1,Belgium,48,"14,805",0.32%
2,Bulgaria,16,"11,985",0.13%
3,Croatia,12,"8,460",0.14%
4,Cyprus,9,"4,230",0.21%


In [246]:
eci_17.head()

,Country,Signatures,Threshold,Percentage
0,Austria,173,"13,395",1.29%
1,Belgium,108,"14,805",0.73%
2,Bulgaria,49,"11,985",0.41%
3,Croatia,60,"8,460",0.71%
4,Cyprus,29,"4,230",0.69%


In [247]:
eci_18.head()

,Country,Signatures,Threshold,Percentage
0,Austria,218,"13,395",1.63%
1,Belgium,392,"14,805",2.65%
2,Bulgaria,"1,961","11,985",16.36%
3,Croatia,294,"8,460",3.48%
4,Cyprus,101,"4,230",2.39%


In [248]:
eci_19.head()

,Country,Signatures,Threshold,Percentage
0,Austria,295,"13,395",2.20%
1,Belgium,213,"14,805",1.44%
2,Bulgaria,22,"11,985",0.18%
3,Croatia,32,"8,460",0.38%
4,Cyprus,10,"4,230",0.24%


In [249]:
eci_20.head()

,Country,Signatures,Threshold,Percentage
0,Austria,36,"13,395",0.27%
1,Belgium,60,"14,805",0.41%
2,Bulgaria,9,"11,985",0.08%
3,Croatia,140,"8,460",1.65%
4,Cyprus,0,"4,230",0.00%


In [250]:
eci_21.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"25,136","13,395",187.65%
1,Belgium,"27,753","14,805",187.46%
2,Bulgaria,"71,444","11,985",596.11%
3,Croatia,"9,362","8,460",110.66%
4,Cyprus,603,"4,230",14.26%


In [251]:
eci_22.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"2,498","13,395",18.65%
1,Belgium,"2,880","14,805",19.45%
2,Bulgaria,"1,005","11,985",8.39%
3,Croatia,"4,551","8,460",53.79%
4,Cyprus,419,"4,230",9.91%


In [252]:
eci_23.head()

,Country,Signatures,Threshold,Percentage
0,Austria,"1,795","13,395",13.40%
1,Belgium,"4,235","14,805",28.61%
2,Bulgaria,131,"11,985",1.09%
3,Croatia,"1,233","8,460",14.57%
4,Cyprus,62,"4,230",1.47%


In [253]:
eci_24.head()

,Country,Signatures,Threshold,Percentage
0,Austria,75,"13,395",0.56%
1,Belgium,121,"14,805",0.82%
2,Bulgaria,17,"11,985",0.14%
3,Croatia,21,"8,460",0.25%
4,Cyprus,6,"4,230",0.14%


In [254]:
eci_25.head()

,Country,Signatures,Threshold,Percentage
0,Austria,34,"13,395",0.25%
1,Belgium,54,"14,805",0.36%
2,Bulgaria,63,"11,985",0.53%
3,Croatia,25,"8,460",0.30%
4,Cyprus,4,"4,230",0.09%


In [255]:
eci_26.head()

,Country,Signatures,Threshold,Percentage
0,Austria,209,"13,395",1.56%
1,Belgium,180,"14,805",1.22%
2,Bulgaria,29,"11,985",0.24%
3,Croatia,41,"8,460",0.48%
4,Cyprus,5,"4,230",0.12%


In [256]:
eci_27.head()

,Country,Signatures,Threshold,Percentage
0,Austria,13,"13,395",0.10%
1,Belgium,13,"14,805",0.09%
2,Bulgaria,1,"11,985",0.01%
3,Croatia,3,"8,460",0.04%
4,Cyprus,10,"4,230",0.24%


In [257]:
worldstat.head()

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Birth Rate,Calling Code,Capital/Major City,Co2-Emissions,...,Out of pocket health expenditure,Physicians per thousand,Population,Population: Labor force participation (%),Tax revenue (%),Total tax rate,Unemployment rate,Urban_population,Latitude,Longitude
0,Afghanistan,60,AF,58.10%,"652,230","323,000",32.49,93.0,Kabul,"8,672",...,78.40%,0.28,"38,041,754",48.90%,9.30%,71.40%,11.12%,"9,797,273",33.939110,67.709953
1,Albania,105,AL,43.10%,"28,748","9,000",11.78,355.0,Tirana,"4,536",...,56.90%,1.20,"2,854,191",55.70%,18.60%,36.60%,12.33%,"1,747,593",41.153332,20.168331
2,Algeria,18,DZ,17.40%,"2,381,741","317,000",24.28,213.0,Algiers,"150,006",...,28.10%,1.72,"43,053,054",41.20%,37.20%,66.10%,11.70%,"31,510,100",28.033886,1.659626
3,Andorra,164,AD,40.00%,468,NaN,7.20,376.0,Andorra la Vella,469,...,36.40%,3.33,"77,142",NaN,NaN,NaN,NaN,"67,873",42.506285,1.521801
4,Angola,26,AO,47.50%,"1,246,700","117,000",40.73,244.0,Luanda,"34,693",...,33.40%,0.21,"31,825,295",77.50%,9.20%,49.10%,6.89%,"21,061,025",-11.202692,17.873887


In [258]:
print("First 5 rows of ECI 01 data:")
print(eci_01.head())

First 5 rows of ECI 01 data:
    Country Signatures Threshold Percentage
0   Austria     10,724    13,395     80.06%
1   Belgium     47,459    14,805    320.56%
2  Bulgaria      2,386    11,985     19.91%
3   Croatia      8,574     8,460    101.35%
4    Cyprus        696     4,230     16.45%


In [259]:
print("First 5 rows of ECI 02 data:")
print(eci_02.head())

First 5 rows of ECI 02 data:
    Country Signatures Threshold Percentage
0   Austria        109    13,395      0.81%
1   Belgium        160    14,805      1.08%
2  Bulgaria         47    11,985      0.39%
3   Croatia         76     8,460      0.90%
4    Cyprus         27     4,230      0.64%


In [260]:
print("First 5 rows of ECI 03 data:")
print(eci_03.head())

First 5 rows of ECI 03 data:
    Country Signatures Threshold Percentage
0   Austria     25,373    13,395    189.42%
1   Belgium     18,313    14,805    123.69%
2  Bulgaria      6,552    11,985     54.67%
3   Croatia     65,963     8,460    779.70%
4    Cyprus      3,228     4,230     76.31%


In [261]:
print("First 5 rows of ECI 04 data:")
print(eci_04.head())

First 5 rows of ECI 04 data:
    Country Signatures Threshold Percentage
0   Austria        156    13,395      1.16%
1   Belgium        257    14,805      1.74%
2  Bulgaria         17    11,985      0.14%
3   Croatia         24     8,460      0.28%
4    Cyprus          8     4,230      0.19%


In [262]:
print("First 5 rows of ECI 05 data:")
print(eci_05.head())

First 5 rows of ECI 05 data:
    Country Signatures Threshold Percentage
0   Austria     20,714    13,395    154.64%
1   Belgium     31,846    14,805    215.10%
2  Bulgaria     14,238    11,985    118.80%
3   Croatia     14,403     8,460    170.25%
4    Cyprus      1,997     4,230     47.21%


In [263]:
print("First 5 rows of ECI 06 data:")
print(eci_06.head())

First 5 rows of ECI 06 data:
    Country Signatures Threshold Percentage
0   Austria        714    13,395      5.33%
1   Belgium        564    14,805      3.81%
2  Bulgaria        122    11,985      1.02%
3   Croatia        472     8,460      5.58%
4    Cyprus         29     4,230      0.69%


In [264]:
print("First 5 rows of ECI 07 data:")
print(eci_07.head())

First 5 rows of ECI 07 data:
    Country Signatures Threshold Percentage
0   Austria        352    13,395      2.63%
1   Belgium        426    14,805      2.88%
2  Bulgaria        185    11,985      1.54%
3   Croatia        217     8,460      2.57%
4    Cyprus         63     4,230      1.49%


In [265]:
print("First 5 rows of ECI 08 data:")
print(eci_08.head())

First 5 rows of ECI 08 data:
    Country Signatures Threshold Percentage
0   Austria        177    13,395      1.32%
1   Belgium        217    14,805      1.47%
2  Bulgaria         50    11,985      0.42%
3   Croatia         64     8,460      0.76%
4    Cyprus         20     4,230      0.47%


In [266]:
print("First 5 rows of ECI 09 data:")
print(eci_09.head())

First 5 rows of ECI 09 data:
    Country Signatures Threshold Percentage
0   Austria      1,139    13,395      8.50%
1   Belgium      2,993    14,805     20.22%
2  Bulgaria        197    11,985      1.64%
3   Croatia      1,115     8,460     13.18%
4    Cyprus         53     4,230      1.25%


In [267]:
print("First 5 rows of ECI 10 data:")
print(eci_10.head())

First 5 rows of ECI 10 data:
    Country Signatures Threshold Percentage
0   Austria        200    13,395      1.49%
1   Belgium        253    14,805      1.71%
2  Bulgaria         36    11,985      0.30%
3   Croatia         44     8,460      0.52%
4    Cyprus          1     4,230      0.02%


In [268]:
print("First 5 rows of ECI 11 data:")
print(eci_11.head())

First 5 rows of ECI 11 data:
    Country Signatures Threshold Percentage
0   Austria        100    13,395      0.75%
1   Belgium        280    14,805      1.89%
2  Bulgaria         21    11,985      0.18%
3   Croatia         16     8,460      0.19%
4    Cyprus          4     4,230      0.09%


In [269]:
print("First 5 rows of ECI 12 data:")
print(eci_12.head())

First 5 rows of ECI 12 data:
    Country Signatures Threshold Percentage
0   Austria        194    13,395      1.45%
1   Belgium        182    14,805      1.23%
2  Bulgaria         84    11,985      0.70%
3   Croatia         57     8,460      0.67%
4    Cyprus          6     4,230      0.14%


In [270]:
print("First 5 rows of ECI 13 data:")
print(eci_13.head())

First 5 rows of ECI 13 data:
    Country Signatures Threshold Percentage
0   Austria      1,133    13,395      8.46%
1   Belgium      1,336    14,805      9.02%
2  Bulgaria        492    11,985      4.11%
3   Croatia        798     8,460      9.43%
4    Cyprus         33     4,230      0.78%


In [271]:
print("First 5 rows of ECI 14 data:")
print(eci_14.head())

First 5 rows of ECI 14 data:
    Country  Signatures Threshold Percentage
0   Austria           9    13,395      0.07%
1   Belgium          15    14,805      0.10%
2  Bulgaria           5    11,985      0.04%
3   Croatia           2     8,460      0.02%
4    Cyprus           2     4,230      0.05%


In [272]:
print("First 5 rows of ECI 15 data:")
print(eci_15.head())

First 5 rows of ECI 15 data:
    Country Signatures Threshold Percentage
0   Austria      4,286    13,395     32.00%
1   Belgium     13,942    14,805     94.17%
2  Bulgaria        227    11,985      1.89%
3   Croatia        424     8,460      5.01%
4    Cyprus        113     4,230      2.67%


In [273]:
print("First 5 rows of ECI 16 data:")
print(eci_16.head())

First 5 rows of ECI 16 data:
    Country Signatures Threshold Percentage
0   Austria         54    13,395      0.40%
1   Belgium         48    14,805      0.32%
2  Bulgaria         16    11,985      0.13%
3   Croatia         12     8,460      0.14%
4    Cyprus          9     4,230      0.21%


In [274]:
print("First 5 rows of ECI 17 data:")
print(eci_17.head())

First 5 rows of ECI 17 data:
    Country Signatures Threshold Percentage
0   Austria        173    13,395      1.29%
1   Belgium        108    14,805      0.73%
2  Bulgaria         49    11,985      0.41%
3   Croatia         60     8,460      0.71%
4    Cyprus         29     4,230      0.69%


In [275]:
print("First 5 rows of ECI 18 data:")
print(eci_18.head())

First 5 rows of ECI 18 data:
    Country Signatures Threshold Percentage
0   Austria        218    13,395      1.63%
1   Belgium        392    14,805      2.65%
2  Bulgaria      1,961    11,985     16.36%
3   Croatia        294     8,460      3.48%
4    Cyprus        101     4,230      2.39%


In [276]:
print("First 5 rows of ECI 19 data:")
print(eci_19.head())

First 5 rows of ECI 19 data:
    Country Signatures Threshold Percentage
0   Austria        295    13,395      2.20%
1   Belgium        213    14,805      1.44%
2  Bulgaria         22    11,985      0.18%
3   Croatia         32     8,460      0.38%
4    Cyprus         10     4,230      0.24%


In [277]:
print("First 5 rows of ECI 20 data:")
print(eci_20.head())

First 5 rows of ECI 20 data:
    Country Signatures Threshold Percentage
0   Austria         36    13,395      0.27%
1   Belgium         60    14,805      0.41%
2  Bulgaria          9    11,985      0.08%
3   Croatia        140     8,460      1.65%
4    Cyprus          0     4,230      0.00%


In [278]:
print("First 5 rows of ECI 21 data:")
print(eci_21.head())

First 5 rows of ECI 21 data:
    Country Signatures Threshold Percentage
0   Austria     25,136    13,395    187.65%
1   Belgium     27,753    14,805    187.46%
2  Bulgaria     71,444    11,985    596.11%
3   Croatia      9,362     8,460    110.66%
4    Cyprus        603     4,230     14.26%


In [279]:
print("First 5 rows of ECI 22 data:")
print(eci_22.head())

First 5 rows of ECI 22 data:
    Country Signatures Threshold Percentage
0   Austria      2,498    13,395     18.65%
1   Belgium      2,880    14,805     19.45%
2  Bulgaria      1,005    11,985      8.39%
3   Croatia      4,551     8,460     53.79%
4    Cyprus        419     4,230      9.91%


In [280]:
print("First 5 rows of ECI 23 data:")
print(eci_23.head())

First 5 rows of ECI 23 data:
    Country Signatures Threshold Percentage
0   Austria      1,795    13,395     13.40%
1   Belgium      4,235    14,805     28.61%
2  Bulgaria        131    11,985      1.09%
3   Croatia      1,233     8,460     14.57%
4    Cyprus         62     4,230      1.47%


In [281]:
print("First 5 rows of ECI 24 data:")
print(eci_24.head())

First 5 rows of ECI 24 data:
    Country Signatures Threshold Percentage
0   Austria         75    13,395      0.56%
1   Belgium        121    14,805      0.82%
2  Bulgaria         17    11,985      0.14%
3   Croatia         21     8,460      0.25%
4    Cyprus          6     4,230      0.14%


In [282]:
print("First 5 rows of ECI 25 data:")
print(eci_25.head())

First 5 rows of ECI 25 data:
    Country Signatures Threshold Percentage
0   Austria         34    13,395      0.25%
1   Belgium         54    14,805      0.36%
2  Bulgaria         63    11,985      0.53%
3   Croatia         25     8,460      0.30%
4    Cyprus          4     4,230      0.09%


In [283]:
print("First 5 rows of ECI 26 data:")
print(eci_26.head())

First 5 rows of ECI 26 data:
    Country Signatures Threshold Percentage
0   Austria        209    13,395      1.56%
1   Belgium        180    14,805      1.22%
2  Bulgaria         29    11,985      0.24%
3   Croatia         41     8,460      0.48%
4    Cyprus          5     4,230      0.12%


In [284]:
print("First 5 rows of ECI 27 data:")
print(eci_27.head())

First 5 rows of ECI 27 data:
    Country  Signatures Threshold Percentage
0   Austria          13    13,395      0.10%
1   Belgium          13    14,805      0.09%
2  Bulgaria           1    11,985      0.01%
3   Croatia           3     8,460      0.04%
4    Cyprus          10     4,230      0.24%


In [285]:
print("First 5 rows of the World Stat dataset:")
print(worldstat.head())

First 5 rows of the World Stat dataset:
       Country Density\n(P/Km2) Abbreviation Agricultural Land( %)  \
0  Afghanistan               60           AF                58.10%   
1      Albania              105           AL                43.10%   
2      Algeria               18           DZ                17.40%   
3      Andorra              164           AD                40.00%   
4       Angola               26           AO                47.50%   

  Land Area(Km2) Armed Forces size  Birth Rate  Calling Code  \
0        652,230           323,000       32.49          93.0   
1         28,748             9,000       11.78         355.0   
2      2,381,741           317,000       24.28         213.0   
3            468               NaN        7.20         376.0   
4      1,246,700           117,000       40.73         244.0   

  Capital/Major City Co2-Emissions  ... Out of pocket health expenditure  \
0              Kabul         8,672  ...                           78.40%   
1 

All 7 petitions contain the same overall structure with 28 rows each, one for each european country + the total signatures row, with 4 columns (country name, signatures, threshold, percentage)

The worldstat dataset contains a much more complex structure with 195 rows, one for each world country, and 35 fields containing different country level statistics

#### Check basic information about the dataset

In [286]:
print("Information about the ECI 01 dataset:")
print(eci_01.info())

Information about the ECI 01 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [287]:
print("Information about the ECI 02 dataset:")
print(eci_02.info())

Information about the ECI 02 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [288]:
print("Information about the ECI 03 dataset:")
print(eci_03.info())

Information about the ECI 03 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [289]:
print("Information about the ECI 04 dataset:")
print(eci_04.info())

Information about the ECI 04 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [290]:
print("Information about the ECI 05 dataset:")
print(eci_05.info())

Information about the ECI 05 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [291]:
print("Information about the ECI 06 dataset:")
print(eci_06.info())

Information about the ECI 06 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [292]:
print("Information about the ECI 07 dataset:")
print(eci_07.info())

Information about the ECI 07 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [293]:
print("Information about the ECI 08 dataset:")
print(eci_08.info())

Information about the ECI 08 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [294]:
print("Information about the ECI 09 dataset:")
print(eci_09.info())

Information about the ECI 09 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [295]:
print("Information about the ECI 10 dataset:")
print(eci_10.info())

Information about the ECI 10 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [296]:
print("Information about the ECI 11 dataset:")
print(eci_11.info())

Information about the ECI 11 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [297]:
print("Information about the ECI 12 dataset:")
print(eci_12.info())

Information about the ECI 12 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [298]:
print("Information about the ECI 13 dataset:")
print(eci_13.info())

Information about the ECI 13 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [299]:
print("Information about the ECI 14 dataset:")
print(eci_14.info())

Information about the ECI 14 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     int64 
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: int64(1), object(3)
memory usage: 1.0+ KB
None


In [300]:
print("Information about the ECI 15 dataset:")
print(eci_15.info())

Information about the ECI 15 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [301]:
print("Information about the ECI 16 dataset:")
print(eci_16.info())

Information about the ECI 16 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [302]:
print("Information about the ECI 17 dataset:")
print(eci_17.info())

Information about the ECI 17 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [303]:
print("Information about the ECI 18 dataset:")
print(eci_18.info())

Information about the ECI 18 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [304]:
print("Information about the ECI 19 dataset:")
print(eci_19.info())

Information about the ECI 19 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [305]:
print("Information about the ECI 20 dataset:")
print(eci_20.info())

Information about the ECI 20 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [306]:
print("Information about the ECI 21 dataset:")
print(eci_21.info())

Information about the ECI 21 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [307]:
print("Information about the ECI 22 dataset:")
print(eci_22.info())

Information about the ECI 22 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [308]:
print("Information about the ECI 23 dataset:")
print(eci_23.info())

Information about the ECI 23 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [309]:
print("Information about the ECI 24 dataset:")
print(eci_24.info())

Information about the ECI 24 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [310]:
print("Information about the ECI 25 dataset:")
print(eci_25.info())

Information about the ECI 25 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [311]:
print("Information about the ECI 26 dataset:")
print(eci_26.info())

Information about the ECI 26 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     object
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: object(4)
memory usage: 1.0+ KB
None


In [312]:
print("Information about the ECI 27 dataset:")
print(eci_27.info())

Information about the ECI 27 dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     28 non-null     object
 1   Signatures  28 non-null     int64 
 2   Threshold   27 non-null     object
 3   Percentage  27 non-null     object
dtypes: int64(1), object(3)
memory usage: 1.0+ KB
None


In [313]:
print("Information about the Worldstat dataset:")
print(worldstat.info())

Information about the Worldstat dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 35 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Country                                    195 non-null    object 
 1   Density
(P/Km2)                            195 non-null    object 
 2   Abbreviation                               188 non-null    object 
 3   Agricultural Land( %)                      188 non-null    object 
 4   Land Area(Km2)                             194 non-null    object 
 5   Armed Forces size                          171 non-null    object 
 6   Birth Rate                                 189 non-null    float64
 7   Calling Code                               194 non-null    float64
 8   Capital/Major City                         192 non-null    object 
 9   Co2-Emissions                              188 non-null  

Most of the numerical variables have been imported as object data types 

due to them containing commas, percentage symbols and other strings within both the scraped petition datasets and the imported worldstat dataset

### Missing values per column

In [314]:
eci_01.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [315]:
eci_02.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [316]:
eci_03.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [317]:
eci_04.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [318]:
eci_05.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [319]:
eci_06.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [320]:
eci_07.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [321]:
eci_08.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [322]:
eci_09.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [323]:
eci_10.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [324]:
eci_11.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [325]:
eci_12.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [326]:
eci_13.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [327]:
eci_14.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [328]:
eci_15.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [329]:
eci_16.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [330]:
eci_17.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [331]:
eci_18.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [332]:
eci_19.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [333]:
eci_20.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [334]:
eci_21.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [335]:
eci_22.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [336]:
eci_23.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [337]:
eci_24.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [338]:
eci_25.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [339]:
eci_26.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [340]:
eci_27.isnull().sum()

Country       0
Signatures    0
Threshold     1
Percentage    1
dtype: int64

In [341]:
worldstat.isnull().sum()

Country                                       0
Density\n(P/Km2)                              0
Abbreviation                                  7
Agricultural Land( %)                         7
Land Area(Km2)                                1
Armed Forces size                            24
Birth Rate                                    6
Calling Code                                  1
Capital/Major City                            3
Co2-Emissions                                 7
CPI                                          17
CPI Change (%)                               16
Currency-Code                                15
Fertility Rate                                7
Forested Area (%)                             7
Gasoline Price                               20
GDP                                           2
Gross primary education enrollment (%)        7
Gross tertiary education enrollment (%)      12
Infant mortality                              6
Largest city                            

each petition contains a missing value in threshold and percentage

this is due to the total signatures row which isnt relevent to the country level data

the world stat datat contains a lot of missing values scattered across almost every column except for country and population density

### Total missing values

In [342]:
total_missing = eci_01.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [343]:
total_missing = eci_02.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [344]:
total_missing = eci_03.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [345]:
total_missing = eci_04.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [346]:
total_missing = eci_05.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [347]:
total_missing = eci_06.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [348]:
total_missing = eci_07.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [349]:
total_missing = eci_08.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [350]:
total_missing = eci_09.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [351]:
total_missing = eci_10.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [352]:
total_missing = eci_11.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [353]:
total_missing = eci_12.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [354]:
total_missing = eci_13.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [355]:
total_missing = eci_14.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [356]:
total_missing = eci_15.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [357]:
total_missing = eci_16.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [358]:
total_missing = eci_17.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [359]:
total_missing = eci_18.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [360]:
total_missing = eci_19.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [361]:
total_missing = eci_20.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [362]:
total_missing = eci_21.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [363]:
total_missing = eci_22.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [364]:
total_missing = eci_23.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [365]:
total_missing = eci_24.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [366]:
total_missing = eci_25.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [367]:
total_missing = eci_26.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [368]:
total_missing = eci_27.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 2


In [369]:
total_missing = worldstat.isna().sum().sum()
print("Total missing values in the dataset:", total_missing)

Total missing values in the dataset: 341


each petition contains a missing value in threshold and percentage

this is due to the total signatures row which isnt relevent to the country level data

the world stat datat contains a lot of missing values scattered across almost every column except for country and population density

total missing values per petition: 2

total missing values in worldstat dataset: 341

### Display column names

In [370]:
print("ECI 01 column names:")
print(eci_01.columns)

ECI 01 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [371]:
print("ECI 02 column names:")
print(eci_02.columns)

ECI 02 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [372]:
print("ECI 03 column names:")
print(eci_03.columns)

ECI 03 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [373]:
print("ECI 04 column names:")
print(eci_04.columns)

ECI 04 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [374]:
print("ECI 05 column names:")
print(eci_05.columns)

ECI 05 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [375]:
print("ECI 06 column names:")
print(eci_06.columns)

ECI 06 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [376]:
print("ECI 07 column names:")
print(eci_07.columns)

ECI 07 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [377]:
print("ECI 08 column names:")
print(eci_08.columns)

ECI 08 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [378]:
print("ECI 09 column names:")
print(eci_09.columns)

ECI 09 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [379]:
print("ECI 10 column names:")
print(eci_10.columns)

ECI 10 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [380]:
print("ECI 11 column names:")
print(eci_11.columns)

ECI 11 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [381]:
print("ECI 12 column names:")
print(eci_12.columns)

ECI 12 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [382]:
print("ECI 13 column names:")
print(eci_13.columns)

ECI 13 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [383]:
print("ECI 14 column names:")
print(eci_14.columns)

ECI 14 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [384]:
print("ECI 15 column names:")
print(eci_15.columns)

ECI 15 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [385]:
print("ECI 16 column names:")
print(eci_16.columns)

ECI 16 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [386]:
print("ECI 17 column names:")
print(eci_17.columns)

ECI 17 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [387]:
print("ECI 18 column names:")
print(eci_18.columns)

ECI 18 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [388]:
print("ECI 19 column names:")
print(eci_19.columns)

ECI 19 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [389]:
print("ECI 20 column names:")
print(eci_20.columns)

ECI 20 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [390]:
print("ECI 21 column names:")
print(eci_21.columns)

ECI 21 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [391]:
print("ECI 22 column names:")
print(eci_22.columns)

ECI 22 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [392]:
print("ECI 23 column names:")
print(eci_23.columns)

ECI 23 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [393]:
print("ECI 24 column names:")
print(eci_24.columns)

ECI 24 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [394]:
print("ECI 25 column names:")
print(eci_25.columns)

ECI 25 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [395]:
print("ECI 26 column names:")
print(eci_26.columns)

ECI 26 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [396]:
print("ECI 27 column names:")
print(eci_27.columns)

ECI 27 column names:
Index(['Country', 'Signatures', 'Threshold', 'Percentage'], dtype='object')


In [397]:
print("Worldstat column names:")
print(worldstat.columns)

Worldstat column names:
Index(['Country', 'Density\n(P/Km2)', 'Abbreviation', 'Agricultural Land( %)',
       'Land Area(Km2)', 'Armed Forces size', 'Birth Rate', 'Calling Code',
       'Capital/Major City', 'Co2-Emissions', 'CPI', 'CPI Change (%)',
       'Currency-Code', 'Fertility Rate', 'Forested Area (%)',
       'Gasoline Price', 'GDP', 'Gross primary education enrollment (%)',
       'Gross tertiary education enrollment (%)', 'Infant mortality',
       'Largest city', 'Life expectancy', 'Maternal mortality ratio',
       'Minimum wage', 'Official language', 'Out of pocket health expenditure',
       'Physicians per thousand', 'Population',
       'Population: Labor force participation (%)', 'Tax revenue (%)',
       'Total tax rate', 'Unemployment rate', 'Urban_population', 'Latitude',
       'Longitude'],
      dtype='object')


The column structure is consistent across the petitions, this will allow for easy combination

the world stat structure is a little more complex and will require careful merging based on the countries in the petition datasets and a petition id 

### Duplicate checking

In [398]:
print("ECI 01:", eci_01.duplicated().sum())
print("ECI 02:", eci_02.duplicated().sum())
print("ECI 03:", eci_03.duplicated().sum())
print("ECI 04:", eci_04.duplicated().sum())
print("ECI 05:", eci_05.duplicated().sum())
print("ECI 06:", eci_06.duplicated().sum())
print("ECI 07:", eci_07.duplicated().sum())
print("ECI 08:", eci_08.duplicated().sum())
print("ECI 09:", eci_09.duplicated().sum())
print("ECI 10:", eci_10.duplicated().sum())
print("ECI 11:", eci_11.duplicated().sum())
print("ECI 12:", eci_12.duplicated().sum())
print("ECI 13:", eci_13.duplicated().sum())
print("ECI 14:", eci_14.duplicated().sum())
print("ECI 15:", eci_15.duplicated().sum())
print("ECI 16:", eci_16.duplicated().sum())
print("ECI 17:", eci_17.duplicated().sum())
print("ECI 18:", eci_18.duplicated().sum())
print("ECI 19:", eci_19.duplicated().sum())
print("ECI 20:", eci_20.duplicated().sum())
print("ECI 21:", eci_21.duplicated().sum())
print("ECI 22:", eci_22.duplicated().sum())
print("ECI 23:", eci_23.duplicated().sum())
print("ECI 24:", eci_24.duplicated().sum())
print("ECI 25:", eci_25.duplicated().sum())
print("ECI 26:", eci_26.duplicated().sum())
print("ECI 27:", eci_27.duplicated().sum())
print("Worldstat:", worldstat.duplicated().sum())

ECI 01: 0
ECI 02: 0
ECI 03: 0
ECI 04: 0
ECI 05: 0
ECI 06: 0
ECI 07: 0
ECI 08: 0
ECI 09: 0
ECI 10: 0
ECI 11: 0
ECI 12: 0
ECI 13: 0
ECI 14: 0
ECI 15: 0
ECI 16: 0
ECI 17: 0
ECI 18: 0
ECI 19: 0
ECI 20: 0
ECI 21: 0
ECI 22: 0
ECI 23: 0
ECI 24: 0
ECI 25: 0
ECI 26: 0
ECI 27: 0
Worldstat: 0


Each dataset contains 0 duplicates

### Initial EDA Observations:

The ECI datasets contain petition data and summary rows that will require cleaning before proper analysis

most variables are stored as text rather than numerical values and will require cleaning and transforming into numericals before statistical analysis

missing values exist in the Worldstat dataset and will require investigation during preparation

country naming conventions may differ between datasets and will need to be standardised before dataset merging

Due to the nature of the uncleaned, unprepared data

Proper Data exploration cannot be conducted until the data has been properly cleaned and prepared

due to most numerical fields containing string data such as commas, and percentage signs

as well as missing values

### Further and extensive Data exploration will be conducted after Data Cleaning and Preparation